daemon/src/maker_cfd_actor.rs

```diff
@@ -207,32 +215,80 @@ impl MakerCfdActor {
         };
 
         // 2. Insert CFD in DB
-        // TODO: Don't auto-accept, present to user in UI instead
         let cfd = Cfd::new(
             current_order.clone(),
-            quantity,
-            CfdState::Accepted {
+            msg.quantity,
+            CfdState::IncomingOrderRequest {
                 common: CfdStateCommon {
                     transition_timestamp: SystemTime::now(),
                 },
+                taker_id,
             },
         );
         insert_cfd(cfd, &mut conn).await?;
 
+        self.cfd_feed_actor_inbox
+            .send(load_all_cfds(&mut conn).await?)?;
+
+        // 3. Remove current order
+        self.current_order_id = None;
+        self.takers()?
+            .do_send_async(maker_inc_connections_actor::BroadcastOrder(None))
+            .await?;
+        self.order_feed_sender.send(None)?;
+
+        Ok(())
+    }
+
+    async fn handle_accept_order(
+        &mut self,
+        msg: AcceptOrder,
+        ctx: &mut Context<Self>,
+    ) -> Result<()> {
+        tracing::debug!(%msg.order_id, "Maker accepts an order" );
+
+        let mut conn = self.db.acquire().await?;
+
+        // Validate if order is still valid
+        let cfd = load_cfd_by_order_id(msg.order_id, &mut conn).await?;
+
+        let taker_id = match cfd {
+            Cfd {
+                state: CfdState::IncomingOrderRequest { taker_id, .. },
+                ..
+            } => taker_id,
+            _ => {
+                anyhow::bail!("Order is in invalid state. Ignoring trying to accept it.")
+            }
+        };
+
+        // Update order in db
+        let order_id = cfd.order.id;
+        insert_new_cfd_state_by_order_id(
+            order_id,
+            CfdState::Accepted {
+                common: CfdStateCommon {
+                    transition_timestamp: SystemTime::now(),
+                },
+            },
+            &mut conn,
+        )
+        .await
+        .unwrap();
+
         self.takers()?
             .do_send_async(maker_inc_connections_actor::TakerMessage {
                 taker_id,
-                command: TakerCommand::NotifyOrderAccepted { id: order_id },
+                command: TakerCommand::NotifyOrderAccepted { id: msg.order_id },
             })
             .await?;
         self.cfd_feed_actor_inbox
             .send(load_all_cfds(&mut conn).await?)?;
 
-        // 3. Remove current order
-        self.current_order_id = None;
         self.takers()?
             .do_send_async(maker_inc_connections_actor::BroadcastOrder(None))
             .await?;
+        self.current_order_id = None;
         self.order_feed_sender.send(None)?;
```

Comment:
Doesn't hurt to do this but I think it may never actually do anything because we set all this to `None` upon `TakeOrder`.